pennylane/grouping/transformations.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Functions for finding the unitary transformations required to diagonalize Pauli partitions in the
+measurement basis, and templates for the circuit implementations.
+"""
+import pennylane as qml
+from pennylane.operation import Tensor
+from pennylane.wires import Wires
+from pennylane.templates import template
+from pennylane.grouping.utils import pauli_to_binary, is_qwc
+import numpy as np
+
+
+@template
+def qwc_rotation(pauli_dict):
+    """Performs circuit implementation of diagonalizing unitary for a Pauli string.
+
+    **Usage example:**
+
+    >>> pauli_dict = {0: qml.PauliX(0), 2: qml.PauliY(2), 3: qml.PauliZ(3)}
```

Comment:
Could be worth adding an example using `Wire` instances as keys

Reply:
Coming back to `qwc_rotation`, I think it might be beneficial to change the input structure. The reason being: a dictionary with `Wires` as keys and single-qubit Pauli operator instances as values seems somewhat redundant and could lead to confusion... an example would be {Wires('a'): PauliX('b')}, for which the rotation will be placed on wire `Wires('a')`, which is not completely intuitive.

Maybe it would be better to just have input as a list of single qubit Pauli operators, e.g. [PauliX('a'), PauliY('b')].